### ISSUE TITLE
go/types, types2: spurious compile error when embedding a parameterized type in a struct

### ISSUE THREAD

**Issue by adonovan** [golang/go#44688]
This program (https://go2goplay.golang.org/p/rF1kHzbn4Fc):
```
type A[T any] struct{}

func (*A[T]) f(T) {}

type B[T any] struct{ A[T] }

func main() {
	var b B[string]
	b.A.f("") // ok
	b.f("") // compile error: type *B[string] of b does not match *A[K] (cannot infer T)
}
```

causes go2go to emit this spurious-looking error:
```
prog.go2:12:2: type *B[string] of b does not match *A[T] (cannot infer T)
```

dev.go2go version 356c1719ec430d55da2bb86dce114926867773f5

**Comment by ianlancetaylor** [comment]
Looks like an error in the type checker.

CC @griesemer

**Comment by griesemer** [comment]
Agreed. This should work.

**Comment by griesemer** [comment]
https://golang.org/cl/298129 fixes this for cmd/compile/internal/types2 and thus the compiler (type checking). Once this is ported to go/types and merged into dev.go2go, it will be fixed there as well (though the actual go2go translator may fail).

**Comment by findleyr** [comment]
Note that this has not yet been merged to dev.go2go, or deployed to the go2go playground.

I'll update here once this is done.

**Comment by findleyr** [comment]
This has been merged and deployed. Now that example just runs into #44689, I believe.
